File: src/main/java/cabalgatas_salento/reservas/mapper/GuiaMapper.java
package cabalgatas_salento.reservas.mapper;

import cabalgatas_salento.reservas.dto.request.GuiaRequest;
import cabalgatas_salento.reservas.dto.response.GuiaResponse;
import cabalgatas_salento.reservas.entity.Guia;
import org.springframework.stereotype.Component;

@Component
public class GuiaMapper {

    public GuiaResponse toResponse(Guia guia) {
        return GuiaResponse.builder()
                .id(guia.getId())
                .primerNombre(guia.getPrimerNombre())
                .primerApellido(guia.getPrimerApellido())
                .tipoDocumento(guia.getTipoDocumento())
                .fechaNacimiento(guia.getFechaNacimiento())
                .documento(guia.getDocumento())
                .telefono(guia.getTelefono())
                .email(guia.getEmail())
                .isActive(guia.getIsActive())
                .eliminado(guia.getEliminado())
                .build();
    }

    public Guia toEntity(GuiaRequest request) {
        return Guia.builder()
                .primerNombre(request.getPrimerNombre())
                .primerApellido(request.getPrimerApellido())
                .tipoDocumento(request.getTipoDocumento())
                .fechaNacimiento(request.getFechaNacimiento())
                .documento(request.getDocumento())
                .telefono(request.getTelefono())
                .email(request.getEmail())
                .build();
    }
}

File: src/guias/mappers/guia.mapper.ts
import { Guia } from '../entities/guia.entity';
import { GuiaResponseDto } from '../dto/guia-response.dto';

export class GuiaMapper {
  static toResponseDto(entity: Guia): GuiaResponseDto {
    const dto = new GuiaResponseDto();
    dto.id = entity.id;
    dto.primerNombre = entity.primerNombre;
    dto.primerApellido = entity.primerApellido;
    dto.tipoDocumento = entity.tipoDocumento;
    dto.fechaNacimiento = entity.fechaNacimiento;
    dto.documento = entity.documento;
    dto.telefono = entity.telefono;
    dto.email = entity.email;
    dto.isActive = entity.isActive;
    dto.eliminado = entity.eliminado;
    return dto;
  }
}
